Document StackTraceLineNumberSupport option (#52989)
dotnet/docs: docs/core/deploying/trimming/trimming-options.md

| `InvariantGlobalization` | When set to `true`, removes globalization-specific code and data. For more information, see [Invariant mode](../../runtime-config/globalization.md#invariant-mode). |
| `MetadataUpdaterSupport` | When set to `false`, removes metadata update&ndash;specific logic related to hot reload. |
| `MetricsSupport` | When set to `false`, removes support for <xref:System.Diagnostics.Metrics> instrumentation. |

> | `StackTraceLineNumberSupport` (.NET 11+) | (`PublishAot` only.) When set to `true`, generates additional information about line numbers into the output executable module. Stack traces (for example, <xref:System.Environment.StackTrace?displayProperty=nameWithType> or <xref:System.Exception.ToString*?displayProperty=nameWithType>) will include information about file names and line numbers at run time. This information is similar to the information generated into debugging symbol files (PDB/DWO/dSYM files), however for apps published with `PublishAot` the runtime doesn't read the native symbol files and the debugging symbols are only used by debuggers. |

Reviewer: ```suggestion
| `StackTraceLineNumberSupport` (.NET 11+) | (`PublishAot` only.) When set to `true`, generates additional line number information in the output executable module. Stack traces (for example, <xref:System.Environment.StackTrace?displayProperty=nameWithType> and <xref:System.Exception.ToString*?displayProperty=nameWithType>) will include information about file names and line numbers at runtime. This information is similar to the information generated into debugging symbol files (PDB/DWO/dSYM files). However, for apps published with `PublishAot`, the runtime doesn't read the native symbol files and the debugging symbols are only used by debuggers. |
```

Author: @gewarren The suggestion to replace "at run time" with "at runtime" seems to be at odds with https://github.com/dotnet/docs/pull/30967#discussion_r960966585. Did the guidance change?

Reviewer: @MichalStrehovsky Yes, the guidance has changed since 2022. See https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/r/runtime.
